common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -1836,6 +1836,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
     HIVEALIAS("hive.alias", "", ""),
     HIVEMAPSIDEAGGREGATE("hive.map.aggr", true, "Whether to use map-side aggregation in Hive Group By queries"),
     HIVEGROUPBYSKEW("hive.groupby.skewindata", false, "Whether there is skew in data to optimize group by queries"),
+    HIVE_JOIN_SHORTCUT_EXTENSIONROWS("hive.join.shortcut.extensionrows", true,
+        "Enables to shortcut processing of known filtered rows in merge joins."),
```

Comment:
If my understanding is correct if this is set to false the outer join produces incorrect results. I don't think it is a good idea to have a property which determines the correctness of the results.

Reply:
idea was to add this toggle to make it possible to check if this feature is causing any perf regression - and make it possible to validate that quickly

Reply:
I see your point but I still think that it would be better to avoid toggles when it comes to correctness issues. I guess it is not the first time we are doing this so I am not gonna push back on this but it shouldn't be the norm. I am leaving the final judgement up to you.

If it goes in I would add some extra information that it is for "internal use only" and changing the value can lead to incorrect results.

Reply:
Agree that this config is needed for performance testing.  At the same time, agree that it should be marked for internal use.  I see there's precedent ..there are a few config options with 'Internal use only' in the description.